### PR TITLE
[chore] Add test coverage and refactor `Address#transfer`

### PR DESCRIPTION
**Pull request by alex-stone**
### What changed? Why?

This adds test coverage around the Address model, specifically the `#transfer` and `#transfers` methods.

This also refactors `Address#transfer` to make it a bit easier to parse in anticipation for transferring using server signer and supporting trading.

#### Qualified Impact

This could cause issues with the Transfer feature

### CODE REVIEW

File: spec/unit/coinbase/wallet_spec.rb [hunk 1]
```diff
@@ -454,32 +454,30 @@
...
       before do
         allow(wallet.default_address)
           .to receive(:transfer)
-          .with(amount, asset_id, to_address_id)
+          .with(amount, asset_id, destination)
           .and_return(transfer)
       end
 
-      it 'creates a transfer to the address ID' do
+      it 'creates a transfer from the defaul address to the address' do
```
Comment (jazz-cb) [inline spec/unit/coinbase/wallet_spec.rb:480]: default

File: lib/coinbase/address.rb [hunk 2]
```diff
@@ -194,5 +159,56 @@ def addresses_api
     def transfers_api
       @transfers_api ||= Coinbase::Client::TransfersApi.new(Coinbase.configuration.api_client)
     end
+
+    def destination_address_and_network(destination)
+      return [destination.default_address.id, destination.network_id] if destination.is_a?(Wallet)
+      return [destination.id, destination.network_id] if destination.is_a?(Address)
+
+      [destination, network_id]
```
Comment (jazz-cb) [inline lib/coinbase/address.rb:167]: shouldnt this be an error?

File: lib/coinbase/address.rb [hunk 3]
```diff
@@ -194,5 +159,56 @@ def addresses_api
...
+      }
+
+      transfer_model = Coinbase.call_api do
+        transfers_api.create_transfer(wallet_id, id, create_transfer_request)
+      end
+
+      Coinbase::Transfer.new(transfer_model)
+    end
+
+    def sign_transfer(transfer)
```
Comment (jazz-cb) [inline lib/coinbase/address.rb:199]: could you add docs for these new methods and document what they return?

Comment (alex-stone) [reply lib/coinbase/address.rb:199]: These are private methods, we can add docs but we would explicitly want them not to be included

File: lib/coinbase/address.rb (hunk 2 above)
Comment (alex-stone) [reply lib/coinbase/address.rb:167]: This is the base case when sending to a string address. Further validation e.g. that the address is formatted properly for the given network will be made server-side

**Review by jazz-cb — APPROVED** [review]

**Comment by alex-stone** [comment]
💇 @jazz-cb 

### Changes
* Fixed typo in wallet spec!

**Review by jazz-cb — APPROVED** [review]

**Pull request merged**